### ISSUE TITLE
request.socket in an https server emits headers via the 'data' event

### ISSUE THREAD

**Issue by jcoglan** [joyent/node#5385]
Compare these two scripts: one starts an HTTP server, makes a TCP connection to it, and sends an Upgrade request. The second starts an HTTPS server, makes a TLS connection and sends the same request.

```js
// logger.js
var Stream = require('stream'),
    util   = require('util');

var Logger = function() {
  this.writable = true;
};
util.inherits(Logger, Stream);

Logger.prototype.write = function(chunk) {
  console.log(chunk.toString());
};

module.exports = Logger;
```

```js
// http.js
var Logger = require('./logger'),
    http   = require('http'),
    net    = require('net');

var server = http.createServer();

server.on('upgrade', function(request, socket, head) {
  console.log('UPGRADE');
  var logger = new Logger();
  socket.pipe(logger);
});

server.listen(8000);

setTimeout(function() {
  var socket = net.connect(8000, 'localhost', {rejectUnauthorized: false}, function() {

    socket.write(new Buffer(
        'GET / HTTP/1.1\r\n' +
        'Host: localhost:8000\r\n' +
        'Upgrade: websocket\r\n' +
        'Connection: Upgrade\r\n' +
        '\r\n',
        'utf8'));
  });
}, 100);
```

```js
// https.js
var Logger = require('./logger'),
    https  = require('https'),
    tls    = require('tls'),
    fs     = require('fs');

var server = https.createServer({
  key:  fs.readFileSync(__dirname + '/spec/server.key'),
  cert: fs.readFileSync(__dirname + '/spec/server.crt')
});

server.on('upgrade', function(request, socket, head) {
  console.log('UPGRADE');
  var logger = new Logger();
  socket.pipe(logger);
});

server.listen(8000);

setTimeout(function() {
  var socket = tls.connect(8000, 'localhost', {rejectUnauthorized: false}, function() {

    socket.write(new Buffer(
        'GET / HTTP/1.1\r\n' +
        'Host: localhost:8000\r\n' +
        'Upgrade: websocket\r\n' +
        'Connection: Upgrade\r\n' +
        '\r\n',
        'utf8'));
  });
}, 100);
```

Under Node v0.10.5, the latter script prints the request headers to stdout, since `socket.pipe()` sends this data to the logger. This only happens for HTTPS servers and only on Node v0.10. I cannot reproduce it with v0.8.

**Comment by jasnell** [comment]
@jcoglan ... is this still an issue?

**Comment by jcoglan** [comment]
It seems that this was fixed in 0.10.19.
